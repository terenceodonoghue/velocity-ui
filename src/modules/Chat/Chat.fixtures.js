export const conversations = {
  data: [
    {
      avatarSrc: './users/lisa.png',
      isOnline: true,
      name: 'Lisa F. Bogar',
      preview: "If you're still having trouble loading...",
      time: '2m',
    },
    {
      avatarSrc: './users/kathey.png',
      isOnline: false,
      name: 'Kathey K. Hernandez',
      preview: 'You are most welcome Kathey!',
      time: '40m',
    },
    {
      avatarSrc: './users/nora.png',
      isOnline: true,
      name: 'Nora M. Buchanan',
      preview: 'I have an issue with...',
      time: '51m',
    },
    {
      avatarSrc: './users/james.png',
      isOnline: false,
      name: 'James A. Parker',
      preview: "Let's look into this together.",
      time: '2h',
    },
    {
      avatarSrc: './users/emma.png',
      isOnline: false,
      name: 'Emma R. Joiner',
      preview: "We're happy to take a look into...",
      time: '7h',
    },
    {
      avatarSrc: './users/martin.png',
      isOnline: false,
      name: 'Martin L. Sherrod',
      preview: 'The download should now start...',
      time: '7h',
    },
    {
      avatarSrc: './users/angel.png',
      isOnline: true,
      name: 'Angel V. Mason',
      preview: 'Sorry to hear about the syncing...',
      time: '10h',
    },
  ],
};

export const dialog = {
  data: [
    {
      avatarSrc: './users/joan.png',
      message:
        'Hello John, thank you for calling Provide Support. How may I help you?',
      name: 'Anna',
      time: '18:54',
    },
    {},
    {
      avatarSrc: './users/joan.png',
      message:
        "I'm sorry, I don't have the answer to that question. May I put you on hold for a few minutes while I check with my manager?",
      name: 'Anna',
      time: '18:54',
    },
    {},
    {},
  ],
};
