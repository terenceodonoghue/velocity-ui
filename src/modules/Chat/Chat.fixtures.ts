import faker from 'faker';

export const conversations = {
  data: [
    {
      isOnline: true,
      message: "If you're still having trouble loading...",
      name: 'Lisa F. Bogar',
      src: faker.image.avatar(),
      time: '2m',
    },
    {
      isOnline: false,
      message: 'You are most welcome Kathey!',
      name: 'Kathey K. Hernandez',
      src: faker.image.avatar(),
      time: '40m',
    },
    {
      isOnline: true,
      message: 'I have an issue with...',
      name: 'Nora M. Buchanan',
      src: faker.image.avatar(),
      time: '51m',
    },
    {
      isOnline: false,
      message: "Let's look into this together.",
      name: 'James A. Parker',
      src: faker.image.avatar(),
      time: '2h',
    },
    {
      isOnline: false,
      message: "We're happy to take a look into...",
      name: 'Emma R. Joiner',
      src: faker.image.avatar(),
      time: '7h',
    },
    {
      isOnline: false,
      message: 'The download should now start...',
      name: 'Martin L. Sherrod',
      src: faker.image.avatar(),
      time: '7h',
    },
    {
      isOnline: true,
      message: 'Sorry to hear about the syncing...',
      name: 'Angel V. Mason',
      src: faker.image.avatar(),
      time: '10h',
    },
  ],
};

export const dialog = {
  data: [
    {
      message:
        'Hello John, thank you for calling Provide Support. How may I help you?',
      name: 'Anna',
      src: faker.image.avatar(),
      time: '18:54',
    },
    {
      message:
        "I'm sorry, I don't have the answer to that question. May I put you on hold for a few minutes while I check with my manager?",
      name: 'Anna',
      src: faker.image.avatar(),
      time: '18:54',
    },
  ],
};
