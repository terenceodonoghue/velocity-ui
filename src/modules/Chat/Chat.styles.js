import styled from '@emotion/styled';
import { Container, Global } from '../../components';

const {
  variables: { colors, effects, typography },
} = Global;

export const Avatar = styled.div`
  flex-grow: 0;
  margin-right: 10px;
`;

export const Body = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  flex-grow: 1;
`;

export const Conversation = styled(Container.Card)`
  display: flex;
  border-left: 0;
  border-top: 0;
  padding: 17px 25px;
  cursor: pointer;
  transition: ${effects.transition};

  &:hover {
    background-color: rgba(46, 91, 255, 0.08);
  }
`;

export const Conversations = styled.ul`
  border-radius: 1px;
  height: 100%;
  margin: 0;
  padding: 0;
  width: 378px;
  background-color: ${colors.white};
  list-style: none;
  overflow: hidden;
`;

export const Details = styled.div`
  display: flex;
  justify-content: space-between;
  line-height: 1.47;
`;

export const Name = styled.span`
  flex-grow: 1;
  font-weight: ${typography.fontWeightMedium};
`;

export const Preview = styled.span`
  display: block;
  color: ${colors.blueyGrey};
  line-height: 1.47;
  max-width: 259px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`;

export const Time = styled.span`
  color: ${colors.blueyGrey};
`;
