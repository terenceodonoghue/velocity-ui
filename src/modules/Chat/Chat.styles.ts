import { css, InterpolationWithTheme, SerializedStyles } from '@emotion/core';
import { Theme } from '../../components';

export const avatar = css`
  flex-grow: 0;
  margin-right: 10px;
`;

export const conversation: InterpolationWithTheme<Theme> = (theme) => css`
  display: flex;
  border-left: 0;
  border-top: 0;
  padding: 17px 25px;
  cursor: pointer;
  transition: ${theme.effects.transition};

  &:hover {
    background-color: rgba(46, 91, 255, 0.08);
  }
`;

export const conversations: InterpolationWithTheme<Theme> = (theme) => css`
  border-radius: 1px;
  height: 100%;
  margin: 0;
  padding: 0;
  width: 378px;
  background-color: ${theme.colors.white};
  list-style: none;
  overflow: hidden;
`;

export const conversationBody = css`
  display: flex;
  flex-direction: column;
  justify-content: center;
  flex-grow: 1;
`;

export const conversationDetails = css`
  display: flex;
  justify-content: space-between;
  line-height: 1.47;
`;

export const conversationName: InterpolationWithTheme<Theme> = (theme) => css`
  flex-grow: 1;
  font-weight: ${theme.typography.fontWeightMedium};
`;

export const conversationPreview: InterpolationWithTheme<Theme> = (
  theme,
) => css`
  display: block;
  color: ${theme.colors.blueyGrey};
  line-height: 1.47;
  max-width: 255px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`;

export const conversationTime: InterpolationWithTheme<Theme> = (theme) => css`
  color: ${theme.colors.blueyGrey};
`;

export const dialog: InterpolationWithTheme<Theme> = (theme) => css`
  background-color: ${theme.colors.white};
`;

export const dialogInput = css`
  margin: 24px;
`;

export const dialogMessages = css`
  padding: 24px;
`;

export const dialogMessage = ({
  isCurrentUser,
}: {
  isCurrentUser: boolean;
}): InterpolationWithTheme<Theme> => (theme): SerializedStyles =>
  css`
    background-color: ${isCurrentUser ? theme.colors.blue : theme.colors.white};
    color: ${isCurrentUser ? theme.colors.white : theme.colors.black};
  `;
