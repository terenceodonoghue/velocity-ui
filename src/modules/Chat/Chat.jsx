import React from 'react';
import { arrayOf, bool, shape, string } from 'prop-types';
import { Avatar } from '../../components';
import * as S from './Chat.styles';

const Conversations = ({ data }) => (
  <S.Conversations>
    {data.map((conversation, index) => (
      <S.Conversation
        // eslint-disable-next-line react/no-array-index-key
        key={`${conversation.name
          .split(' ')
          .join('-')
          .toLowerCase()}-${index}`}
      >
        <S.Avatar>
          <Avatar.Round src={conversation.avatarSrc} height="48" width="48" />
        </S.Avatar>
        <S.Body>
          <S.Details>
            <S.Name>{conversation.name}</S.Name>
            <S.Time>{conversation.time}</S.Time>
          </S.Details>
          <S.Preview>{conversation.preview}</S.Preview>
        </S.Body>
      </S.Conversation>
    ))}
  </S.Conversations>
);

const Dialog = ({ data }) => (
  <S.Dialog>
    <S.Messages>
      {data.map((message) => (
        <S.Message>{JSON.stringify(message)}</S.Message>
      ))}
    </S.Messages>
    <S.TextInput />
  </S.Dialog>
);

Conversations.propTypes = {
  data: arrayOf(
    shape({
      avatarSrc: string.isRequired,
      isOnline: bool.isRequired,
      name: string.isRequired,
      preview: string.isRequired,
      time: string.isRequired,
    }).isRequired,
  ).isRequired,
};

Dialog.propTypes = {
  data: arrayOf(
    shape({
      avatarSrc: string.isRequired,
      message: string.isRequired,
      name: string.isRequired,
      time: string.isRequired,
    }).isRequired,
  ).isRequired,
};

export default {
  Conversations,
  Dialog,
};
