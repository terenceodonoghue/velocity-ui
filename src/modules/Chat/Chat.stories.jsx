import React from 'react';
import { storiesOf } from '@storybook/react';
import { Conversations, Dialog } from './Chat';
import * as fixtures from './Chat.fixtures';

storiesOf('Modules|Chat', module)
  .add('Conversations', () => <Conversations {...fixtures.conversations} />)
  .add('Dialog', () => <Dialog {...fixtures.dialog} />);
