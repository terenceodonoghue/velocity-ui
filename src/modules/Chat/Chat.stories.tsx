import React, { FunctionComponent } from 'react';
import * as Chat from './Chat';
import * as fixtures from './Chat.fixtures';

export default {
  title: 'Modules/Chat',
};

export const Conversations: FunctionComponent = () => (
  <Chat.Conversations {...fixtures.conversations} />
);
export const Dialog: FunctionComponent = () => (
  <Chat.Dialog {...fixtures.dialog} />
);
