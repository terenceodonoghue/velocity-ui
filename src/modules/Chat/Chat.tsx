/** @jsx jsx */
import { jsx } from '@emotion/core';
import { FunctionComponent } from 'react';
import { Avatars, Inputs, Sendable, Surfaces } from '../../components';
import * as css from './Chat.styles';

interface Conversation extends Sendable {
  isOnline: boolean;
}

interface Message extends Sendable {
  isCurrentUser: boolean;
}

interface ConversationsProps {
  data: Conversation[];
}

interface DialogProps {
  data: Message[];
}

export const Conversations: FunctionComponent<ConversationsProps> = ({
  data,
}) => (
  <ul css={css.conversations}>
    {data.map((conversation, index) => (
      <Surfaces.Card
        css={css.conversation}
        // eslint-disable-next-line react/no-array-index-key
        key={`${conversation.name.split(' ').join('-').toLowerCase()}-${index}`}
      >
        <div css={css.avatar}>
          <Avatars.Round src={conversation.src} height="48" width="48" />
        </div>
        <div css={css.conversationBody}>
          <div css={css.conversationDetails}>
            <span css={css.conversationName}>{conversation.name}</span>
            <span css={css.conversationTime}>{conversation.time}</span>
          </div>
          <span css={css.conversationPreview}>{conversation.message}</span>
        </div>
      </Surfaces.Card>
    ))}
  </ul>
);

export const Dialog: FunctionComponent<DialogProps> = ({ data }) => (
  <div css={css.dialog}>
    {data.map((message) => (
      <div
        css={css.dialogMessage({
          isCurrentUser: message.isCurrentUser,
        })}
      >
        <span css={css.dialogName({ isCurrentUser: message.isCurrentUser })}>
          {message.name}
        </span>
        {message.message}
        <span css={css.dialogTime({ isCurrentUser: message.isCurrentUser })}>
          {message.time}
        </span>
        <img
          alt={message.name}
          css={css.dialogAvatar({ isCurrentUser: message.isCurrentUser })}
          src={message.src}
        />
      </div>
    ))}
    <Inputs.Text css={css.dialogInput} />
  </div>
);
