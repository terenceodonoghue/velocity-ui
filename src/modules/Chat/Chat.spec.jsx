/** @jsx jsx */
import { jsx } from '@emotion/core';
import { Conversations } from './Chat';
import * as fixtures from './Chat.fixtures';

describe('Chat', () => {
  describe('Conversations', () => {
    it('matches snapshot', () => {
      const tree = renderer
        .create(withTheme(<Conversations {...fixtures.conversations} />))
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
