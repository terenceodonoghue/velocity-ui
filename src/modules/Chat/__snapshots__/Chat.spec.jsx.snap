// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat Conversations matches snapshot 1`] = `
.emotion-56 {
  border-radius: 1px;
  height: 100%;
  margin: 0;
  padding: 0;
  width: 378px;
  background-color: #fff;
  list-style: none;
  overflow: hidden;
}

.emotion-7 {
  display: block;
  border-radius: 1px;
  border: solid 1px rgba(46,91,255,0.08);
  box-shadow: 0 10px 20px 0 rgba(46,91,255,0.07);
  padding: 32px 24px;
  background-color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-left: 0;
  border-top: 0;
  padding: 17px 25px;
  cursor: pointer;
  -webkit-transition: all 0.2s linear;
  transition: all 0.2s linear;
}

.emotion-7:hover {
  background-color: rgba(46,91,255,0.08);
}

.emotion-1 {
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  margin-right: 10px;
}

.emotion-0 {
  border-radius: 50%;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  line-height: 1.47;
}

.emotion-2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-weight: 500;
}

.emotion-3 {
  color: #8798ad;
}

.emotion-5 {
  display: block;
  color: #8798ad;
  line-height: 1.47;
  max-width: 259px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

<ul
  className="emotion-56"
>
  <div
    className="emotion-7"
  >
    <div
      className="emotion-1"
    >
      <img
        className="emotion-0"
        height="48"
        src="https://s3.amazonaws.com/uifaces/faces/twitter/xravil/128.jpg"
        width="48"
      />
    </div>
    <div
      className="emotion-6"
    >
      <div
        className="emotion-4"
      >
        <span
          className="emotion-2"
        >
          Lisa F. Bogar
        </span>
        <span
          className="emotion-3"
        >
          2m
        </span>
      </div>
      <span
        className="emotion-5"
      >
        If you're still having trouble loading...
      </span>
    </div>
  </div>
  <div
    className="emotion-7"
  >
    <div
      className="emotion-1"
    >
      <img
        className="emotion-0"
        height="48"
        src="https://s3.amazonaws.com/uifaces/faces/twitter/vladyn/128.jpg"
        width="48"
      />
    </div>
    <div
      className="emotion-6"
    >
      <div
        className="emotion-4"
      >
        <span
          className="emotion-2"
        >
          Kathey K. Hernandez
        </span>
        <span
          className="emotion-3"
        >
          40m
        </span>
      </div>
      <span
        className="emotion-5"
      >
        You are most welcome Kathey!
      </span>
    </div>
  </div>
  <div
    className="emotion-7"
  >
    <div
      className="emotion-1"
    >
      <img
        className="emotion-0"
        height="48"
        src="https://s3.amazonaws.com/uifaces/faces/twitter/iduuck/128.jpg"
        width="48"
      />
    </div>
    <div
      className="emotion-6"
    >
      <div
        className="emotion-4"
      >
        <span
          className="emotion-2"
        >
          Nora M. Buchanan
        </span>
        <span
          className="emotion-3"
        >
          51m
        </span>
      </div>
      <span
        className="emotion-5"
      >
        I have an issue with...
      </span>
    </div>
  </div>
  <div
    className="emotion-7"
  >
    <div
      className="emotion-1"
    >
      <img
        className="emotion-0"
        height="48"
        src="https://s3.amazonaws.com/uifaces/faces/twitter/safrankov/128.jpg"
        width="48"
      />
    </div>
    <div
      className="emotion-6"
    >
      <div
        className="emotion-4"
      >
        <span
          className="emotion-2"
        >
          James A. Parker
        </span>
        <span
          className="emotion-3"
        >
          2h
        </span>
      </div>
      <span
        className="emotion-5"
      >
        Let's look into this together.
      </span>
    </div>
  </div>
  <div
    className="emotion-7"
  >
    <div
      className="emotion-1"
    >
      <img
        className="emotion-0"
        height="48"
        src="https://s3.amazonaws.com/uifaces/faces/twitter/giancarlon/128.jpg"
        width="48"
      />
    </div>
    <div
      className="emotion-6"
    >
      <div
        className="emotion-4"
      >
        <span
          className="emotion-2"
        >
          Emma R. Joiner
        </span>
        <span
          className="emotion-3"
        >
          7h
        </span>
      </div>
      <span
        className="emotion-5"
      >
        We're happy to take a look into...
      </span>
    </div>
  </div>
  <div
    className="emotion-7"
  >
    <div
      className="emotion-1"
    >
      <img
        className="emotion-0"
        height="48"
        src="https://s3.amazonaws.com/uifaces/faces/twitter/johnsmithagency/128.jpg"
        width="48"
      />
    </div>
    <div
      className="emotion-6"
    >
      <div
        className="emotion-4"
      >
        <span
          className="emotion-2"
        >
          Martin L. Sherrod
        </span>
        <span
          className="emotion-3"
        >
          7h
        </span>
      </div>
      <span
        className="emotion-5"
      >
        The download should now start...
      </span>
    </div>
  </div>
  <div
    className="emotion-7"
  >
    <div
      className="emotion-1"
    >
      <img
        className="emotion-0"
        height="48"
        src="https://s3.amazonaws.com/uifaces/faces/twitter/_kkga/128.jpg"
        width="48"
      />
    </div>
    <div
      className="emotion-6"
    >
      <div
        className="emotion-4"
      >
        <span
          className="emotion-2"
        >
          Angel V. Mason
        </span>
        <span
          className="emotion-3"
        >
          10h
        </span>
      </div>
      <span
        className="emotion-5"
      >
        Sorry to hear about the syncing...
      </span>
    </div>
  </div>
</ul>
`;
